src/ipyvizzu/chart.py

```diff
@@ -1,153 +1,76 @@
 """
-Jupyter notebook integration for Vizzu.
+Jupyter Notebook integration of Vizzu.
 """
-
-import pkgutil
-import uuid
-
 from IPython.display import display_javascript
 from IPython import get_ipython
 
-from ipyvizzu.animation import Animation, Snapshot, AnimationMerger
-from ipyvizzu.method import Animate, Feature, Store
-from ipyvizzu.template import DisplayTarget, DisplayTemplate
-
+from pyvizzu.chart import Chart as PyvizzuChart
 
-class Chart:
-    """
-    Wrapper over Vizzu Chart
-    """
+from ipyvizzu.animation import Snapshot
+from ipyvizzu.template import DisplayTarget, DisplayTemplate, VIZZU
 
-    VIZZU = "https://cdn.jsdelivr.net/npm/vizzu@~0.4.0/dist/vizzu.min.js"
 
+class Chart(PyvizzuChart):
     def __init__(
         self,
         vizzu=VIZZU,
         width="800px",
         height="480px",
         display: DisplayTarget = DisplayTarget("actual"),
     ):
-        self._chart_id = uuid.uuid4().hex[:7]
+        self._js = {}
+        self._js["target"] = DisplayTarget(display)
 
-        self._display_target = DisplayTarget(display)
-        self._calls = []
-        self._showed = False
+        self._classes = {}
+        self._classes["DisplayTemplate"] = DisplayTemplate
+        self._classes["Snapshot"] = Snapshot
 
-        self._scroll_into_view = False
+        super().__init__(vizzu, width, height)
```

Comment:
We should call `super().__init__(...)` first and set the `_classes` and `_js` values on the parent's data.

However it seems `ipyvizzu` and `stpyvizzu` use `ipyvizzu.template.DisplayTemplate` and `ipyvizzu.animation.Snapshot`. Do we need this parameterization?

Reply:
@nyirog You are right, streamlit-pyvizzu uses pyvizzu's DisplayTemplate and Snapshot, so streamlit-pyvizzu's SnapShot does not need to be parameterized.
But ipyvizzu still needs a different DisplayTemplate and because of that, a parameterized Snapshot, (Snapshot using DisplayTemplate).

The constructor is almost the same in pyvizzu and ipyvizzu, but ipyvizzu's DisplayTemplate (and Snapshot) are different, so I set them before the parent's constructor called.

Because we call _set_chart() in the constructor and _set_chart needs DisplayTemplate and DisplayTarget, we need to set these in the child before super called, or we can place _set_chart into a different public method, but it brakes the current behavior. What do you think?

Reply:
I would like to avoid to initialize the same attribute from two `__init__`, and it would be better to avoid the usage of the protected attributes of the parent class.

The `DisplayTemplate` and `Snapshot` could be a parameter of `PyvizzuChart` with default values, or `PyvizzuChart` could have `_get_display_template_class` and  `_get_snapshot_class` methods which can be overwritten by `ipyvizzu.chart.Chart`.

Reply:
@nyirog Thanks, I'll refactor this based on your suggestions and I'll add _get_display_template_class/snapshot_class methods.

Reply:
If you prefer `@property`, then `_display_template_class` and `_snapshot_class` could be a property of `PyvizzuChart` which is overwritten by `ipyvizzu.chart.Chart`.

Reply:
> Is it acceptable to set property befor calling super init?
Yes, it may be the most readable implementation. However we have some other options, too.

If the _display_target is only configured in the constructor, then we can use class attribute / property.

We can configure the class attribute with inheritence:
 ```python
class Chart:
   _display_target = ...

class ManualChart(Chart):
  _display_target = MANUAL
```

Or with class method:
```python
class Chart:
  @classmethod
  def with_display_target(cls, vizzu, width, height, display_target):
    cls._display_target = display_target
    return  cls(vizzu, width, height)
```

The inheritence is better if we have only 2  or 3 option for display_target.

Ps.: Sorry for the late answer. I had no time in the weekend:(